gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java

```diff
@@ -417,6 +417,7 @@ private void  handleLeadershipChange(NotificationContext changeContext) {
         //Activate DagManager only if TopologyCatalog is initialized. If not; skip activation.
         if (this.topologyCatalog.getInitComplete().getCount() == 0) {
           this.dagManager.setActive(true);
+          this.eventBus.register(this.dagManager);
```

Comment:
Any chance we can update GobblinServiceHATest to verify that on leadership change the dagManager has been registered with the eventbus? e.g. construct a mock DagManager and pass it to GobblinServiceManager for it to register the mock dag manager?